cmd/buildkitd/config/gcpolicy_windows.go

```diff
@@ -3,10 +3,36 @@
 
 package config
 
+import "golang.org/x/sys/windows"
+
 func DetectDefaultGCCap() DiskSpace {
-	return DiskSpace{Bytes: defaultCap}
+	return DiskSpace{Percentage: 10}
 }
 
 func (d DiskSpace) AsBytes(root string) int64 {
-	return d.Bytes
+	if d.Bytes != 0 {
+		return d.Bytes
+	}
+	if d.Percentage == 0 {
+		return 0
+	}
+
+	rootUTF16, err := windows.UTF16FromString(root)
+	if err != nil {
+		return defaultCap
+	}
+	var freeAvailableBytes uint64
+	var totalBytes uint64
+	var totalFreeBytes uint64
+
+	if err := windows.GetDiskFreeSpaceEx(
+		&rootUTF16[0],
+		&freeAvailableBytes,
+		&totalBytes,
+		&totalFreeBytes); err != nil {
+		return defaultCap
```

Comment:
q: wondering if in debug mode, whether we should print out this error if it ever happens (though has less likelihood)? I see on `gcpolicy_unix.go`, we are silently failing.

Reply:
I think we can log it as warning even when not in debug mode.